src/components/TradeWidget/OrderValidity.tsx

```diff
@@ -58,137 +82,211 @@ const Wrapper = styled.div`
+  // )
+
+  const handleShowConfig = useCallback((e): void => (e.preventDefault(), setShowOrderConfig(!showOrderConfig)), [
+    setShowOrderConfig,
+    showOrderConfig,
+  ])
+  const handleShowOrderConfig = useCallback((e): void => (e.preventDefault(), setShowOrderConfig(false)), [
+    setShowOrderConfig,
+  ])
+
+  const handleValidFromChange = useCallback(() => {
+    const newValue = adjustPrecision(validFromInputValue, 0)
+    if (validFromInputValue !== newValue) {
+      setValue(validFromInputId, newValue, true)
+    }
+  }, [validFromInputValue, setValue, validFromInputId])
+
+  const handleValidUntilChange = useCallback(() => {
+    const newValue = adjustPrecision(validUntilInputValue, 0)
+    if (validUntilInputValue !== newValue) {
+      setValue(validUntilInputId, newValue, true)
     }
-  }, [inputValue, setValue, inputId])
+  }, [validUntilInputValue, setValue, validUntilInputId])
 
   useEffect(() => {
-    handleChange()
-  }, [handleChange])
+    handleValidFromChange()
+  }, [handleValidFromChange])
 
-  // function handleUnlimitedClick(): void {
-  //   setUnlimited(!isUnlimited)
-  //   !isUnlimited ? setValue(inputId, '', true) : setValue(inputId, '30', true)
-  // }
+  useEffect(() => {
+    handleValidUntilChange()
+  }, [handleValidUntilChange])
+
+  function handleUnlimitedClick(): void {
+    setUnlimited(!isUnlimited)
+    !isUnlimited ? setValue(validUntilInputId, '', true) : setValue(validUntilInputId, '30', true)
+  }
+  function handleASAPClick(): void {
+    setAsap(!isAsap)
+    !isAsap ? setValue(validFromInputId, '', true) : setValue(validFromInputId, '30', true)
+  }
 
   //TODO: re-enable input logic
   return (
     <Wrapper>
-      <button>
-        Order starts: <b>ASAP</b> - expires in: <b>30 minutes</b>
+      <button onClick={handleShowConfig}>
+        Order starts: <b>{formatValidity(validFrom, 'ASAP')}</b> - expires in: <b>{formatValidity(validUntil, '-')}</b>
       </button>
-      {/* <InputBox>
-        <div className="main-input-container">
-          <input
-            className={className}
-            name={inputId}
-            type="number"
-            step="5"
-            disabled={isDisabled || isUnlimited}
-            required
-            ref={register({
-              pattern: { value: validInputPattern, message: 'Expiration time cannot be negative' },
-              validate: { positive: validatePositive },
-            })}
-            onChange={handleChange}
-            onFocus={(e): void => e.target.select()}
-            tabIndex={tabIndex + 2}
-          />
-          <div className="radio-container">
-            <input type="checkbox" disabled={isDisabled} defaultChecked={isUnlimited} onClick={handleUnlimitedClick} />
-            <small>Never</small>
-          </div>
-        </div>
-        {errorOrWarning}
-      </InputBox> */}
+
+      <OrderValidityInputsWrapper $visible={showOrderConfig}>
```

Comment:
Would be better to conditionally render here instead of switching visibility
If it breaks form-hooks somehow, try putting the whole thing + useFormContext into its own component and conditionally rendering that

Reply:
why would it break hooks?

Reply:
no idea
just thinking

Reply:
@anxolin @alfetopito thoughts on this? - I'm fine with it using the opactiy as its not a huge component but i get the sentiment from Dima

Reply:
It's ok to just hide, but you can't just use opacity
If you really want to, at least add `pointer-events: none`, otherwise your modal steals clicks

Reply:
ah true, didn't think of that. ok I'll work it in

Reply:
No opinion, so I'll just back @Velenir

Reply:
I'm ok with hidding too. What I'm confused about is that this part looks like is the implementation of a modal (I might be wrong), while we had a modal component already. So theoretically we didn't need to re-style this again and handle the visibility thing. It's handled in the library

I think this probably has a relation to the nit-issue reported here: https://github.com/gnosis/dex-react/issues/629